Source/Frontend/UI/UI_VanguardImplementation.cs

```diff
@@ -472,5 +150,453 @@ private static void OnMessageReceived(object sender, NetCoreEventArgs e)
+        private static void PushCorruptCoreSpecUpdate(NetCoreAdvancedMessage advancedMessage, ref NetCoreEventArgs e)
+        {
+            SyncObjectSingleton.FormExecute(() =>
+                        {
+                            RTCV.NetCore.AllSpec.CorruptCoreSpec?.Update((PartialSpec)advancedMessage.objectValue, false);
+                        });
+            e.setReturnValue(true);
+        }
+
+        private static void GenerateVmdText(NetCoreAdvancedMessage advancedMessage, ref NetCoreEventArgs e)
+        {
+            SyncObjectSingleton.FormExecute(() =>
+            {
+                object[] objs = (object[])advancedMessage.objectValue;
+                string domain = (string)objs[0];
+                string text = (string)objs[1];
+
+                var vmdgenerator = S.GET<RTC_VmdGen_Form>();
+
+                vmdgenerator.btnSelectAll_Click(null, null);
+
+                var cbitems = vmdgenerator.cbSelectedMemoryDomain.Items;
+                object domainFound = null;
+                for (int i = 0; i < cbitems.Count; i++)
+                {
+                    var item = cbitems[i];
+
+                    if (item.ToString() == domain)
+                    {
+                        domainFound = item;
+                        vmdgenerator.cbSelectedMemoryDomain.SelectedIndex = i;
+                        break;
+                    }
+                }
+
+                if (domainFound == null)
+                {
+                    throw new Exception($"Domain {domain} could not be selected in the VMD Generator. Aborting procedure.");
+                                            //return;
+                                        }
+
+                vmdgenerator.tbCustomAddresses.Text = text;
+
+                string value = "";
+
+                if (RTCV.UI.UI_Extensions.GetInputBox("VMD Generation", "Enter the new VMD name:", ref value) == DialogResult.OK)
+                {
+                    if (!string.IsNullOrWhiteSpace(value))
+                        vmdgenerator.tbVmdName.Text = value.Trim();
+                    vmdgenerator.btnGenerateVMD_Click(null, null);
+                }
+            });
+            e.setReturnValue(true);
+        }
+
+        private static void DomainsUpdated()
+        {
+            SyncObjectSingleton.FormExecute(() =>
+            {
+                S.GET<RTC_MemoryDomains_Form>().RefreshDomains();
+                S.GET<RTC_MemoryDomains_Form>().SetMemoryDomainsAllButSelectedDomains(RTCV.NetCore.AllSpec.VanguardSpec[VSPEC.MEMORYDOMAINS_BLACKLISTEDDOMAINS] as string[] ?? new string[] { });
+            });
+        }
+
+        private static void GetBlastGeneratorLayer(ref NetCoreEventArgs e)
+        {
+            BlastLayer bl = null;
+            SyncObjectSingleton.FormExecute(() =>
+            {
+                bl = S.GET<RTC_BlastGenerator_Form>().GenerateBlastLayers(true, true, false);
```

Comment:
I had to modify this function since `ref NetCoreEventArgs e` can't be used inside of an `Action`